controllers/database_reconciler_helpers.go

```diff
@@ -222,12 +222,18 @@ func (r *DatabaseReconciler) handleSync(ctx context.Context, database *ndbv1alph
 
 		databaseAdapter := &controller_adapters.Database{Database: *database}
 		generatedReq, err := ndb_api.GenerateProvisioningRequest(ctx, ndbClient, databaseAdapter, reqData)
-		// generatedReq, err := ndb_api.GenerateProvisioningRequestt(ctx, ndbClient, database.Spec, reqData)
+
 		if err != nil {
-			log.Error(err, "Could not generate provisioning request, requeuing.")
+			log.Error(err, "Could not generate provisioning request, requeuing...")
 			return r.requeueOnErr(err)
 		}
```

Comment:
[Non Blocking/General Discussion] - Since you're already working on this file/improvement: How do you feel about moving the request generation into the ProvisionDatabase call itself?